### DISCUSSION TITLE
Studio: port-forward fails connecting to pods with IPv6 only listeners

### DISCUSSION THREAD

**Discussion by RaphaelBossek** [supabase/supabase#22191] [Questions]
I've followed the tutorial for self-hosting https://supabase.com/docs/guides/self-hosting for Stackgres on Microsoft Azure Kubernates Services (K8s 1.27.9) and local KIND and failing with following problem:

```
kubectl port-forward --address 0.0.0.0 svc/demo-private-supabase-studio 3000
Forwarding from 0.0.0.0:3000 -> 3000
Handling connection for 3000
Handling connection for 3000
E0323 16:47:51.012559   19660 portforward.go:409] an error occurred forwarding 3000 -> 3000: error forwarding port 3000 to pod 4276b646033804fc7d05569ba5c951bc0a994ba8ac391f3ae5624d34acac3419, uid : failed to execute portforward in network namespace "/var/run/netns/cni-0be6d99d-f331-73c5-1d9c-ccd0bff9ff0f": failed to connect to localhost:3000 inside namespace "4276b646033804fc7d05569ba5c951bc0a994ba8ac391f3ae5624d34acac3419", IPv4: dial tcp4 127.0.0.1:3000: connect: connection refused IPv6 dial tcp6: address localhost: no suitable address found
E0323 16:47:51.014626   19660 portforward.go:409] an error occurred forwarding 3000 -> 3000: error forwarding port 3000 to pod 4276b646033804fc7d05569ba5c951bc0a994ba8ac391f3ae5624d34acac3419, uid : failed to execute portforward in network namespace "/var/run/netns/cni-0be6d99d-f331-73c5-1d9c-ccd0bff9ff0f": failed to connect to localhost:3000 inside namespace "4276b646033804fc7d05569ba5c951bc0a994ba8ac391f3ae5624d34acac3419", IPv4: dial tcp4 127.0.0.1:3000: connect: connection refused IPv6 dial tcp6: address localhost: no suitable address found
error: lost connection to pod
```

As far I understand the topic it's related to https://github.com/kubernetes/kubernetes/issues/99850

The last release of Supabase Studio that works for me is **20231030-9abc686** (https://github.com/supabase/supabase/pkgs/container/studio). All new releases are affected by this issue.

**Comment by William-LP** [comment]
Hey @RaphaelBossek, I know it's been a while but did you manage to get this to works with some of the latest release?

**Comment by 1000p** [comment]
If you really, really need to do that you can set HOSTNAME to 0.0.0.0. That will enable you to port forward.
It will probably mess up other stuff down the chain, but who knows 🤷‍♂️  Maybe someone with more knowledge in the ecosystem can help here.

There is an [issue for that that is closed currently](https://github.com/supabase/supabase/issues/29812)
